lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -121,13 +123,34 @@ def #{configuration[:column]}=(position)
 
             after_commit 'remove_instance_variable(:@scope_changed) if defined?(@scope_changed)'
 
-            scope :in_list, lambda { where(%q{#{quoted_table_name}.#{connection.quote_column_name(configuration[:column])} IS NOT NULL}) }
+            scope :in_list, lambda { where(%q{#{quoted_table_name}.#{quoted_position_column} IS NOT NULL}) }
+
+            def self.decrement_all_with_touch
+              update_all_with_touch %q(#{quoted_position_column} = (#{quoted_position_column} - 1))
+            end
+
+            def self.increment_all_with_touch
+              update_all_with_touch %q(#{quoted_position_column} = (#{quoted_position_column} + 1))
+            end
           EOV
 
           if configuration[:add_new_at].present?
             self.send :before_create, :"add_to_list_#{configuration[:add_new_at]}"
           end
+        end
+
+        private
 
+        def update_all_with_touch(updates)
+          record = new
+          attrs = record.send(:timestamp_attributes_for_update_in_model)
+          now = record.send(:current_time_from_proper_timezone)
+
+          query = attrs.map { |attr| "#{attr} = :now" }
```

Comment:
Good catch. I took a look through the AR source, ran some tests, and I have concluded that we don't need to, because even without the check, this function's behavior is consistent with `#touch`. 

`#empty?` is truthy in four circumstances: `nil`, `""`, `[]`, and `{}`. If someone sets `#timestamp_attributes_for_update_in_model` to `nil` or `""`, AR will throw exceptions trying to call `#each` on it when saving, so we'd be pretty consistent with that behavior by throwing an exception trying to call `#map` on it. For the latter two circumstances, `#map` nicely takes care of emptiness by returning `[]` for both, which works well with the remainder of the method.